File: Medium/to_camel_case.py
# https://edabit.com/challenge/4bX7qAqELq9XJeB26
def to_camel_case(txt):
    if txt:
        f,s,t=txt.split("_") if "_" in txt else txt.split("-")
        return "{}{}{}".format(f,s.capitalize(),t.capitalize())
    return ""



print(to_camel_case("the_stealth_warrior"), "theStealthWarrior", "Did not return correct value")
print(to_camel_case("The-Stealth-Warrior"), "TheStealthWarrior", "Did not return correct value")
print(to_camel_case("A-B-C"), "ABC", "Did not return correct value.")
print(to_camel_case(''), '', "An empty string was provided but not returned.")

File: javascript/Medium/toCamelCase.js
// https://www.codewars.com/kata/517abf86da9663f1d2000003/train/javascript

function toCamelCase(str){
    return str.split(/\-|\_/g).map((el,i)=> {
        if(!i&&str.slice(0,1).toUpperCase()!==str.slice(0,1)) return el

        return el.slice(0,1).toUpperCase()+el.slice(1)
    }).join('')
}

// better solution
function toCamelCase(str){
    return str.replace(/[-_](.)/g, (_, c) => c.toUpperCase());
  }

console.log(toCamelCase(''), '', "An empty string was provided but not returned")
console.log(toCamelCase("the_stealth_warrior"), "theStealthWarrior", "toCamelCase('the_stealth_warrior') did not return correct value")
console.log(toCamelCase("The-Stealth-Warrior"), "TheStealthWarrior", "toCamelCase('The-Stealth-Warrior') did not return correct value")
console.log(toCamelCase("A-B-C"), "ABC", "toCamelCase('A-B-C') did not return correct value")
